python/cudf/cudf/core/column/lists.py

```diff
@@ -451,3 +457,56 @@ def sort_values(
             sort_lists(self._column, ascending, na_position),
             retain_index=not ignore_index,
         )
+
+    def flatten(self) -> ParentType:
+        """
+        Removes one level of nesting from each row of the list Series.
+
+        Returns
+        -------
+        Series or Index with one level of nesting removed from each row.
+
+        Examples
+        --------
+        >>> s1
+        0      [[1.0, 2.0], [3.0, 4.0, 5.0]]
+        1    [[6.0, nan], [7.0], [8.0, 9.0]]
+        dtype: list
+        >>> s1.list.flatten()
+        0    [1.0, 2.0, 3.0, 4.0, 5.0]
+        1    [6.0, nan, 7.0, 8.0, 9.0]
+        dtype: list
+
+        Null values at the top-level in each row are dropped:
+
+        >>> s2
+        0    [[1.0, 2.0], None, [3.0, 4.0, 5.0]]
+        1        [[6.0, nan], [7.0], [8.0, 9.0]]
+        dtype: list
+        >>> s2.list.flatten()
+        0    [1.0, 2.0, 3.0, 4.0, 5.0]
+        1    [6.0, nan, 7.0, 8.0, 9.0]
+        dtype: list
+        """
+        result_dtype = self._column.dtype.element_type
+        if not isinstance(result_dtype, ListDtype):
+            return self._return_or_inplace(self._column)
+
+        self_offsets = self._column.children[0]
+        child_offsets = self._column.children[1].children[0]
```

Comment:
Unrelated to this PR but we should really consider building `offsets` and `character_bytes` accessors instead of always accessing the children by index.

Reply:
We have them - I just wasn't using them. Fixed!